### PR TITLE
Huber Loss Function for both kind of behavior. 

### PR DESCRIPTION
**Pull request by ashutosh1919**
Sometimes, we have data in which we want to use multiple loss functions. For small scale data, we want to use MSE and for higher scale data we want to use MAE. The behaviour is not possible with one of these alone. So, here I am including robust loss function **huber** which takes into account a parameter called **huber_delta** which is by default 0.1. In below image, you can see that huber will perform as MSE when residual is less than huber_delta and will perform as MAE when residual is greater or equal to huber_delta.
huber_formula

### CODE REVIEW

File: keras/losses.py [hunk 1]
```diff
@@ -39,6 +39,12 @@ def hinge(y_true, y_pred):
     return K.mean(K.maximum(1. - y_true * y_pred, 0.), axis=-1)
 
 
+def huber(y_true, y_pred, huber_delta=0.1):
+    x = K.abs(y_true - y_pred)
+    x = K.switch(x < huber_delta, 0.5 * x**2, huber_delta * (x - 0.5 * huber_delta))
```
Comment (abhinavsagar) [inline keras/losses.py]: Could you use a different variable here instead of x. Don't think relational operator is supported between instances of 'Function' and 'float' value.

**Comment by ashutosh1919** [comment]
I have changed the code, and now, code is compatible to all backend types and also resolved issues related to it.
Please merge it if you find it useful.

File: keras/losses.py (hunk 1 above)
Comment (ashutosh1919) [reply keras/losses.py]: I have changed the code, and now, code is compatible to all backend types and also resolved issues related to it.
Please merge it if you find it useful.

**Comment by fchollet** [comment]
Thanks for the PR. This type of addition should go into the `tf-2` branch, which is where the next release will come from.

We already have the huber loss in tf-2: https://github.com/keras-team/keras/blob/tf-2/keras/losses.py#L669

You can open a PR to edit it if you see anything missing.
